### PR TITLE
Upgrade nts 2.4

### PR DESCRIPTION
**Pull request by marcridler**
Upgrade NetTopologySuite to 2.4.0 removing all references to GeoAPI. 

--> Breaking Change.

### CODE REVIEW

File: src/DHI.Mesh.sln [hunk 1]
```diff
@@ -22,28 +22,28 @@ Global
...
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
-		{CB530B16-5FB8-44F3-BE57-45D29C1F0E89}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{CB530B16-5FB8-44F3-BE57-45D29C1F0E89}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{CB530B16-5FB8-44F3-BE57-45D29C1F0E89}.Debug|Any CPU.ActiveCfg = Release|Any CPU
+		{CB530B16-5FB8-44F3-BE57-45D29C1F0E89}.Debug|Any CPU.Build.0 = Release|Any CPU
 		{CB530B16-5FB8-44F3-BE57-45D29C1F0E89}.Debug|x64.ActiveCfg = Debug|x64
 		{CB530B16-5FB8-44F3-BE57-45D29C1F0E89}.Debug|x64.Build.0 = Debug|x64
-		{CB530B16-5FB8-44F3-BE57-45D29C1F0E89}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{CB530B16-5FB8-44F3-BE57-45D29C1F0E89}.Release|Any CPU.Build.0 = Release|Any CPU
+		{CB530B16-5FB8-44F3-BE57-45D29C1F0E89}.Release|Any CPU.ActiveCfg = Debug|Any CPU
```
Comment (JesperGr) [inline src/DHI.Mesh.sln]: Seems like Debug builds Release, and Release builds Debug?

File: src/DHI.Mesh.Test/DHI.Mesh.Test.csproj [hunk 2]
```diff
@@ -10,7 +10,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.11.0" />
     <PackageReference Include="DHI.DFS" Version="20.0.0" />
-    <PackageReference Include="NetTopologySuite.IO.GeoJSON" Version="1.15.0" />
+    <PackageReference Include="NetTopologySuite.IO.GeoJSON" Version="2.0.4" />
```
Comment (JesperGr) [inline src/DHI.Mesh.Test/DHI.Mesh.Test.csproj:13]: This is 2.0.4 and not 2.4. Does these not follow each other?

**Comment by JesperGr** [comment]
I think we need to have the versions of NetTopologySuite match the ones in MIKE Zero, MO and Domain services.  For the coming release that may be updated to 2.0.?? So, I would not pull this into master.

**Comment by marcridler** [comment]
I reverted the .sln file to the original one from the master branch. That got changed by accident. The NetTopologySuite.IO.GeoJSON are also upgraded. 

@JesperGr If I understand you correctly, the MIKE builds targets the latest master? So you cannot specify a particular nuget package version? If so, yes Pulling this to master would be a problem.
@larsmichael  If it's a problem to put this in master, could it be a PreRelease? And I can then work on the MikeCore provider.

File: src/DHI.Mesh.Test/DHI.Mesh.Test.csproj (hunk 2 above)
Comment (larsmichael) [reply src/DHI.Mesh.Test/DHI.Mesh.Test.csproj:13]: No, they don't follow each other. Also, this is only used in the test-project anyway.

**Comment by larsmichael** [comment]
Even if 2.4.0 is probably backward compatible with 2.0.0 (at least the semantic versioning would indicate this) it is probably better to be safe than sorry and use 2.0.0. There might also be some strong-named assemblies somewhere forced to use a specific version of NTS. History shows that with commonly used dependencies (like NTS) we are better of using the same version across various components and products. Currently we strive for 2.0.0, so I [modified to use version 2.0.0](https://github.com/DHI/DHI.Mesh/pull/12/commits/0f646aa37ed52d3bb2554e54b8f8323733156abf).

**Comment by larsmichael** [comment]
> ... I would not pull this into master.

I see no problem in doing this. Besides the DfsUtil, the artifact of this repository is the DHI.Mesh nuget package that we will give a version indicating a breaking change (2.0.0)

**Review by larsmichael — APPROVED** [review]

**Comment by larsmichael** [comment]
@marcridler For now, I pushed a version 2.0.0-rc01 to our internal NuGet feed, so you can test it from the MIKECoreProvider.

**Pull request merged**